src/pulp/solvers.py

```diff
@@ -1275,75 +1278,94 @@ def actualSolve(self, lp):
             tmpSol = lp.name+"-pulp.prt"
         lp.writeLP(tmpLp, writeSOS = 1, mip = self.mip)
+        xpress = subprocess.Popen([self.path, lp.name], shell=True, stdin=subprocess.PIPE, universal_newlines=True)
         if not self.msg:
-            xpress = os.popen(self.path+" "+lp.name+" > /dev/null 2> /dev/null", "w")
-        else:
-            xpress = os.popen(self.path+" "+lp.name, "w")
-        xpress.write("READPROB "+tmpLp+"\n")
+            xpress.stdin.write("OUTPUTLOG=0\n")
+        xpress.stdin.write("READPROB "+tmpLp+"\n")
         if self.maxSeconds:
-            xpress.write("MAXTIME=%d\n" % self.maxSeconds)
+            xpress.stdin.write("MAXTIME=%d\n" % self.maxSeconds)
         if self.targetGap:
-            xpress.write("MIPRELSTOP=%f\n" % self.targetGap)
+            xpress.stdin.write("MIPRELSTOP=%f\n" % self.targetGap)
         if self.heurFreq:
-            xpress.write("HEURFREQ=%d\n" % self.heurFreq)
+            xpress.stdin.write("HEURFREQ=%d\n" % self.heurFreq)
         if self.heurStra:
-            xpress.write("HEURSTRATEGY=%d\n" % self.heurStra)
+            xpress.stdin.write("HEURSTRATEGY=%d\n" % self.heurStra)
         if self.coverCuts:
-            xpress.write("COVERCUTS=%d\n" % self.coverCuts)
+            xpress.stdin.write("COVERCUTS=%d\n" % self.coverCuts)
         if self.preSolve:
-            xpress.write("PRESOLVE=%d\n" % self.preSolve)
+            xpress.stdin.write("PRESOLVE=%d\n" % self.preSolve)
         for option in self.options:
-            xpress.write(option+"\n")
+            xpress.stdin.write(option+"\n")
         if lp.sense == LpMaximize:
-            xpress.write("MAXIM\n")
+            xpress.stdin.write("MAXIM\n")
         else:
-            xpress.write("MINIM\n")
+            xpress.stdin.write("MINIM\n")
         if lp.isMIP() and self.mip:
-            xpress.write("GLOBAL\n")
-        xpress.write("WRITEPRTSOL "+tmpSol+"\n")
-        xpress.write("QUIT\n")
-        if xpress.close() != None:
+            xpress.stdin.write("GLOBAL\n")
+        if self.fullPrecision:
+            xpress.stdin.write("WRITESOL "+tmpSol+ " -p\n")
+        else:
+            xpress.stdin.write("WRITEPRTSOL "+tmpSol+"\n")
+        xpress.stdin.write("QUIT\n")
+        xpress.stdin.close()
+        if xpress.wait() != 0:
             raise PulpSolverError("PuLP: Error while executing "+self.path)
         status, values = self.readsol(tmpSol)
         if not self.keepFiles:
             try: os.remove(tmpLp)
             except: pass
             try: os.remove(tmpSol)
             except: pass
+        lp.status = status
         lp.assignVarsVals(values)
-        # TODO: the following could be put inside the assignStatus function
         if abs(lp.infeasibilityGap(self.mip)) > 1e-5: # Arbitrary
-            status = LpStatusInfeasible
-        lp.assignStatus(status)
-        return status
+            lp.status = LpStatusInfeasible
+        return lp.status
```

Comment:
you need to keep lp.assignStatus here instead of directly assigning lp.status a value